UIforETW/Settings.cpp

```diff
@@ -143,11 +143,15 @@ void CSettings::OnBnClickedCopystartupprofile()
 	std::wstring source = exeDir_ + fileName;
 
 	wchar_t documents[MAX_PATH];
-	if (!SHGetSpecialFolderPath(*this, documents, CSIDL_MYDOCUMENTS, TRUE))
+
+	const BOOL getMyDocsResult = SHGetSpecialFolderPathW(m_hWnd, documents, CSIDL_MYDOCUMENTS, TRUE);

```

Comment:
> On the other hand, what would "encapsulation" look like here?

A getHwnd() function and no way to set it. i.e.; getter, but no setter. Preventing uncontrolled changes gives some wonderful guarantees.